Assets/Scripts/Enemies/MeleeFodderEnemy.cs

```diff
@@ -106,12 +113,14 @@ IEnumerator ChargeAndAttack(Transform player)
         if(dashSound != null) dashSound.PlaySound(0, AudioSourceType.Enemy);
 
         Vector3 forward = transform.forward;
+        if (trail != null) { trail.SetActive(true); }
```

Comment:
Functionality works

for polish it would be good to disable the trail after a delay , this makes sure the trail renderer has faded out, and this prevents it to cut out abruptly when the enmey finishes  the charge or attacks player on collsion

`Invoke(nameof(DisableTrailRenderer), 0.5f);` // call this wherever you are disabling the trail

```
void DisableTrailRenderer()
{
    if (trail != null){ trail.SetActive(false); }
}
```